xbmc/interfaces/legacy/WindowInterceptor.h

```diff
@@ -120,7 +120,7 @@ namespace XBMCAddon
       CGUIWindow* get() override { return this; }
 
       // this is only called from XBMC core and we only want it to return true every time
-      virtual bool Update(const String &strPath) { return true; }
+      virtual bool Update(const String&, bool = true) { return true; }
```

Comment:
override is now missing

Reply:
i tried to explain this in the pr text. this template is instanced both for a CGUIWindow and for a CGUIMediaWindow. for the former there is no method to override.

Reply:
have overlooked the commit message

Reply:
Perhaps it would be better to add a default `virtual Update` to CGUIWindow that just returns true and then mark it as override?

Reply:
i considered it but i kinda dislike inflating api due to user code concerns like this. i concluded i prefer the warning, no strong opinion though.

Reply:
my main question was whether or not it can be put in the specific instance only (ie put in WindowXMLInterceptor). do you ever need the "middle layer" ?

Reply:
(it builds fine, but i'm not sure whether there is some runtime stuff happening which depends on the base or not).

Reply:
Wow. This is a mess. I don't understand why this worked without this change.

The interceptor (which should probably be called "dispatcher") acts like a traffic cop for calls going between the scripting language proxies (Window, WindowXML, etc) and the Kodi windowing system (CGUIWindow, CGUIMediaWindow, etc) directing calls either 'up', to the Kodi windowing side, or 'over' to the scripting language side depending on the context of the call.  The additional parameter was already added on the windowing side (CGUIMediaWindow) but not in the dispatcher which is required to override it to work correctly.

So this SHOULD have resulted in runtime issues. Not just a compiler warning.

Reply:
This is probably good to go but let me do a little more digging later today.

Reply:
@notspiff 

1) is it possible to remove this method from Interceptor all together? It will still be on `WindowXMLInterceptor` which is defined in a C++ file and bridges `WindowXML` to `CGUIMediaWindow`. `Window` doesn't define an `Update` and neither does `CGUIWindow` so it's not exposed to Python at that point anyway.

2) It compiles with the deleted.

3) The scripting language side currently isn't getting the additional `bool` parameter (see `WindowXML.h` the `WindowXML` class). So the `WindowXMLInterceptor` is bridging `CGUIMediaWindow::Update(String&, bool)` <-> `WindowXML::Update(String&)`. Should the additional parameter be available to the addon developers?

Reply:
1) that's what i tried to asked above.

Reply:
Personally I'd say delete it but I'm only about 80% sure it wont have an unforeseen affect. Also, you should probably decide on (3) and if you think addon developers should see the additional parameter, add it to WindowXML.

Reply:
if you say so, that's what i will do. i don't want to expose the filtering bool, it's a quirk thingie for internal updates and as i said i doubt the filtering is exposed in general.